### PR TITLE
Factoring out setPhi and getPhi

### LINKED ISSUE #393: Dihedral angle calculation is inconsistent
At https://github.com/Acellera/htmd/blob/master/htmd/parameterization/ffevaluate.py
```python
    def evaluateTorsion(coords, phis):
        pos1 = coords[0, :]
        pos2 = coords[1, :]
        pos3 = coords[2, :]
        pos4 = coords[3, :]

        r12 = pos1 - pos2
        r23 = pos2 - pos3
        r34 = pos3 - pos4

        r12 = r12.flatten()
        r23 = r23.flatten()
        r34 = r34.flatten()

        A = np.cross(r12, r23)
        B = np.cross(r23, r34)
        C = np.cross(r23, A)

        rA = 1. / np.linalg.norm(A)
        rB = 1. / np.linalg.norm(B)
        rC = 1. / np.linalg.norm(C)

        B = B.dot(rB)

        cos_phi = A.dot(B) * rA
        sin_phi = C.dot(B) * rC

        phi = - math.atan2(sin_phi, cos_phi)
...
```

At https://github.com/Acellera/htmd/blob/master/htmd/parameterization/phi.py
```python
def getPhi(coords, atoms):
    coords = np.atleast_3d(coords)
    pos1 = coords[atoms[0], :, 0]
    pos2 = coords[atoms[1], :, 0]
    pos3 = coords[atoms[2], :, 0]
    pos4 = coords[atoms[3], :, 0]

    r12 = pos1 - pos2
    r23 = pos2 - pos3
    r34 = pos3 - pos4

    A = np.cross(r12, r23)
    B = np.cross(r23, r34)
    C = np.cross(r23, A)

    rA = 1. / np.linalg.norm(A)
    rB = 1. / np.linalg.norm(B)
    rC = 1. / np.linalg.norm(C)

    B = B * rB

    cos_phi = (np.dot(A, B)) * rA
    sin_phi = (np.dot(C, B)) * rC

    phi = - math.atan2(sin_phi, cos_phi)
...
```

Identical code, except:
```python
B = B.dot(rB)
```
versus
```python
B = B * rB
```

### PR DESCRIPTION
**Pull request by raimis**
- [x] Fix #393
- [x] Replace `setPhi` with `Molecule.setDihedral`
- [x] Replace `getPhi` with `Molecule.getDihedral`

### CODE REVIEW

File: htmd/parameterization/phi.py [hunk 1]
```diff
@@ -108,37 +108,26 @@ def setPhi(coords, atoms, left, right, phi):
 
 def getPhi(coords, atoms):
```
Comment (stefdoerr) [inline htmd/parameterization/phi.py]: If we are leaving the `atoms` parameter free, I would just remove this function as it doesn't really return phi. It returns any dihedral angle given specific atoms. So I'd just remove the function and call it directly as `dihedralAngle(coords[atoms, :, 0])`

Comment (stefdoerr) [inline htmd/parameterization/phi.py]: Can you remove the `getPhi` function and replace the calls with `dihedralAngle(mol.coords[dihedral, :, 0])` so we can merge this PR?

**Comment by raimis** [comment]
@stefdoerr Wait, this won't go before #423!

**Comment by raimis** [comment]
@stefdoerr 

There is `Molecule.setDihedaral`. Why there is no `getDihedral` method, but rather a separate `dihedralAngle` function?

**Comment by stefdoerr** [comment]
Lost in development. I wrote setDihedral after dihedralAngle for some
purpose and didn't consider consolidating the other one. I will add it now

**Comment by raimis** [comment]
Good! It will streamline API.

**Comment by stefdoerr** [comment]
https://github.com/Acellera/htmd/commit/8ebbf52f38935a8510eacc69f1e6689239ae1b8c

Here you go. Take care though as it returns radians instead of degrees (unlike dihedralAngle) to be consistent with the set method.

**Comment by raimis** [comment]
Thanks @stefdoerr

**Comment by j3mdamas** [comment]
@stefdoerr, don't forget you can do changes on the PRs yourself

**Comment by raimis** [comment]
Not it depends on #397

**Comment by raimis** [comment]
@stefdoerr can you review this, please.

File: htmd/parameterization/ffmolecule.py [hunk 2]
```diff
@@ -405,51 +405,51 @@ def getSoftTorsions(self):
...
-        step = 10  # degrees
-        nstep = int(360 / step)
-        cset = np.zeros((self.natoms, 3, nstep))
-
-        i = 0
-        for phi in range(-180, 180, step):
-            cset[:, :, i] = setPhi(self.coords[:, :, 0], atoms, left, right, phi)
-            i += 1
+        # Number of rotamers for each dihedral to compute
+        nrotemer = 36
```
Comment (stefdoerr) [inline htmd/parameterization/ffmolecule.py]: I mean, if you care about typos, it's nrot*a*mer not nrot*e*mer

**Comment by stefdoerr** [comment]
Other than the typo it's fine. You want to fix it quick?

**Pull request merged**